### DISCUSSION TITLE
Diff artifacts between two runs?

### DISCUSSION THREAD

**Discussion by krikru** [mlflow/mlflow#14047] [Q&A]
Is there any way to display a diff for all artifacts that are logged, between two runs, to see exactly what changed in the experiment setup and what could have caused the difference in model performance (except for varying luck when initializing the weights)? I'm thinking something like [this](https://github.com/mlflow/mlflow/commit/master).

**Comment by krikru** [comment]
I turned this Q&A discussion into an [idea](https://github.com/mlflow/mlflow/discussions/14058).

**Comment by zakariajaadi** [comment]
Hi Krikru, 

Mlflow does actually offer some functionality that can help you manually compare runs, In the MLflow UI, you can select multiple runs and view their parameters and metrics in a table format. This allows you to manually compare values side by side.
However, if you're looking to have a GitHub-like diff functionality for artifacts, I believe mlflow is not designed for this, because it assumes that code versioning is handled by a dedicated Version Control System like Git. 

Hope this answers your question! 

Regards, 
Zakaria
